samtranslator/model/sam_resources.py

```diff
@@ -489,6 +489,7 @@ class SamApi(SamResourceMacro):
     referable_properties = {
         "Stage": ApiGatewayStage.resource_type,
         "Deployment": ApiGatewayDeployment.resource_type,
+        "DomainName": ApiGatewayDomainName.resource_type
     }
```

Comment:
@keetonian You can refer the domainname created with `Api.DomainName` because of this line here.